### ISSUE TITLE
Jackson + Afterburner Module preventing the serialization of null properties

### ISSUE THREAD

**Issue by cchacin** [dropwizard/dropwizard#1627]
**[Repository with tests to reproduce the issue:](https://github.com/cchacin/jackson-afterburner-dw-issue)**
Having this pojo:

```java
public final class PojoResponse {

    private final String        string;
    private final UUID          uuid;
    private final LocalDateTime localDateTime;
    private final boolean       aBoolean;
    private final Boolean       aBooleanT;

    @JsonCreator
    public PojoResponse(@JsonProperty("string") final String string,
                        @JsonProperty("uuid") final UUID uuid,
                        @JsonProperty("localDateTime") final LocalDateTime localDateTime,
                        @JsonProperty("aBoolean") final boolean aBoolean,
                        @JsonProperty("aBooleanT") final Boolean aBooleanT) {
        this.string = string;
        this.uuid = uuid;
        this.localDateTime = localDateTime;
        this.aBoolean = aBoolean;
        this.aBooleanT = aBooleanT;
    }

    public String getString() {
        return this.string;
    }

    public UUID getUuid() {
        return this.uuid;
    }

    public LocalDateTime getLocalDateTime() {
        return this.localDateTime;
    }

    public boolean isaBoolean() {
        return this.aBoolean;
    }

    public Boolean getaBooleanT() {
        return this.aBooleanT;
    }
}
```

When we serialize it, with and without the afterburner module we expect this json on both cases:

```json
{
  "string":null,
  "uuid":null,
  "localDateTime":null,
  "aBoolean":true,
  "aBooleanT":false
}
```

But with the afterburner module this json is the output:

```json
{
  "string":null,
  "aBoolean":true,
  "aBooleanT":false
}
```

**Comment by joschi** [comment]
@cchacin Thanks for this bug report!

It seems related to FasterXML/jackson-modules-base#7 which has been fixed in Jackson 2.8.0.

@dropwizard/committers Any thoughts on updating Jackson for Dropwizard 1.0.0?

**Comment by jplock** [comment]
If the tests all pass, I think it'd be safe to update. We just have to be prepared for a DW 1.0.1 in case a new Jackson is released.

**Comment by cchacin** [comment]
@jplock @joschi do you have a release date for 1.0.0?

**Comment by jplock** [comment]
@cchacin @joschi was working on it in a branch (https://github.com/dropwizard/dropwizard/tree/issue-1627), but I haven't see a pull request yet.  Assuming that passes, I think 1.0.0 should be clear after that.  🚢 :shipit:  🚀

**Comment by joschi** [comment]
@jplock Unfortunately Jackson 2.8.0 changed some internals which make more changes necessary. The `issue-1627` currently doesn't pass: https://travis-ci.org/dropwizard/dropwizard/builds/143862696

Not sure if it's worth postponing the release of Dropwizard 1.0.0 for this.

**Comment by jplock** [comment]
@joschi in that case maybe we just ship 1.0.0 with this as a known issue?

**Comment by pkwarren** [comment]
It sounds like this patch will also be included in 2.7.6 which should be less risk at this point for the 1.0.0 release: https://github.com/FasterXML/jackson/wiki/Jackson-Release-2.7.6

**Comment by cowtowncoder** [comment]
@joschi I agree in that 2.8 probably not a good change quite yet; but curious whether any serious test failures remain in just-released 2.8.1.
I noticed there are some failures for changed exception messages, but as far as I know these are things are just differences in wording. I would be interested in knowing if other failures still occur; could not quite figure out an easy way to skip/comment out these fails.

As to 2.7.6, I'll see if I can get it out by this weekend, so within next 3 days or so.

**Comment by joschi** [comment]
@cowtowncoder Thank you for your feedback! Unfortunately the build still fails with Jackson 2.8.1: https://travis-ci.org/#L4820-L4826

I currently don't have time left to debug this thoroughly so we'll probably wait for Jackson 2.7.6 to be released.

**Comment by cowtowncoder** [comment]
@joschi agree wrt 2.76. But I think the link you have is wrong; would be interested in seeing actual failure. Maybe I can help with that for future version compatibility.

**Comment by joschi** [comment]
@cowtowncoder You are right, the link was wrong. The correct link with the output of the failed Travis CI job is: https://travis-ci.org/dropwizard/dropwizard/builds/146628857#L4717-L4740

**Comment by cowtowncoder** [comment]
2.7.6 is now released.

**Comment by cowtowncoder** [comment]
@joschi I think those errors are with Jackson 2.8.0. I can see other failures with 2.8.1, ones about exception messages, but not the subtype problem which I think is fixed with 2.8.1.

**Comment by cowtowncoder** [comment]
Oh. Hmmh, no. I had only gotten to `config` tests, not core. You are right, failures still with 2.8.1. Rats.

**Comment by joschi** [comment]
> @joschi I think those errors are with Jackson 2.8.0. I can see other failures with 2.8.1, ones about exception messages, but not the subtype problem which I think is fixed with 2.8.1.

Please see #1647.
